src/engagement_db_to_analysis/engagement_db_to_analysis.py

```diff
@@ -111,6 +111,8 @@ def generate_analysis_files(user, google_cloud_credentials_file_path, pipeline_c
     if pipeline_config.analysis.google_drive_upload is None:
         log.debug("Not uploading to Google Drive, because the 'google_drive_upload' configuration was None")
     else:
+        if dry_run:
+            exit(0)
```

Comment:
Prefer
```suggestion
    if pipeline_config.analysis.google_drive_upload is None or dry_run:
        log.debug("Not uploading to Google Drive, because the 'google_drive_upload' configuration was None")
    else:
```
so we don't have to think about possible early exits if we add any code after this block.